### ISSUE TITLE
Slack integration for Marathon deployments

### ISSUE THREAD

**Issue by ssk2** [mesosphere/marathon#1141]
It'd be excellent to have a Slack plugin that sends messages when deployments begin and complete for Marathon applications.

**Comment by rasputnik** [comment]
Nice idea! 

That sounds like something you could easily implement as a web hook using the existing event handler callbacks.

**Comment by lukaswelte** [comment]
Would be great if the person who implements this would not create a slack specific endpoint. 
A generic web hook that is triggered would make a lot more integrations possible out of the box ;)

**Comment by marcust** [comment]
+1

**Comment by dontrebootme** [comment]
Probably easier now with the marathon event bus/subscriptions?

**Comment by rodrickbrown** [comment]
Any update on this request? The slack integration in Chronos has been super useful for us and I would love to see this in Marathon.

**Comment by assertrandom** [comment]
+1

**Comment by tobilg** [comment]
I created a small Node.js project which can do this if run via Marathon on a Mesos or DC/OS cluster: https://github.com/tobilg/marathon-slack

**Comment by jdef** [comment]
Looks cool. Is there a way to rate-limit events that are posted to slack?

**Comment by tobilg** [comment]
@jdef I thought about it but didn't really come up with an idea on the limiting criteria. Would love to hear some ideas...! I think it would be quite easy to implement probably.

**Comment by pyronicide** [comment]
@ssk2 -1 to this being a plugin. I don't want my mission critical infrastructure to have contact with an external service that isn't necessarily dependable. 

@tobilg is definitely going in the right direction. We've got the primitives to build something like this.

**Comment by ssk2** [comment]
@pyronicide: yeah, not sure what I was thinking when I suggested it was a plugin. 

@tobilg: this is great work! Thank you! I'd love to see it available in the DC/OS Universe at some point - so that users can do `dcos package install slack-notifier` or similar. In fact, I want to set this up on our production cluster now!

**Comment by karlkfi** [comment]
Slack already has a great ecosystem for running bots. You can run small ones here for free: https://beepboophq.com/

Of course, you can also run them in docker on marathon in your cluster. I'd recommend that over being some sort of plugin. At some point we'll want to get something like the Heroku button that lets you easily deploy a github/dockerhub project to DC/OS. In the mean time, a package would be cool, and not too hard to write.

**Comment by tobilg** [comment]
@ssk2 Thanks! I created a PR at https://github.com/mesosphere/universe/pull/678. It's my first package, so I hope everything is as it should be.

@karlkfi As I tried to keep things simple, it's not a real "bot". Instead, it just forwards the events it receives from Marathon to a specific Slack Webhook. From my point of view it wouldn't make sense to use an external platform for this. There's no interaction possible from Slack towards the project (and subsequently the DC/OS cluster), as I think this would also have security implications.
